airflow_dbt/hooks/google.py

```diff
@@ -0,0 +1,146 @@
+import logging
+from typing import Any, Dict, List
+
+from airflow.exceptions import AirflowException
+from airflow.providers.google.cloud.hooks.cloud_build import CloudBuildHook
+from airflow.providers.google.cloud.hooks.gcs import GCSHook, _parse_gcs_url
+from airflow.providers.google.get_provider_info import get_provider_info
+from airflow.settings import json
+from packaging import version
+
+from airflow_dbt.hooks.base import DbtBaseHook
+
+# Check we're using the right google provider version. As composer is the
+# most brad used Airflow installation we will default to the latest version
+# composer is using
+google_providers_version = get_provider_info().get('versions')[0]
+v_min = version.parse('5.0.0')
+v_max = version.parse('6.0.0')
+v_provider = version.parse(google_providers_version)
+if not v_min <= v_provider < v_max:
+    raise Exception(
+        'The provider "apache-airflow-providers-google" version "'
+        f'{google_providers_version}" is not compatible with the current API. '
+        f'Please install a compatible version in the range [{v_min}, {v_max})"'
+    )
+
+
+class DbtCloudBuildHook(DbtBaseHook):
+    """
+    Runs the dbt command in a Cloud Build job in GCP
+
+    :type env: dict
+    :param env: If set, passed to the dbt executor
+    :param project_id: GCP Project ID as stated in the console
+    :type project_id: str
+    :param gcp_conn_id: The connection ID to use when fetching connection info.
+    :type gcp_conn_id: str
+    :param gcs_staging_location: Where to store the sources to be fetch later
+        by the cloud build job. It should be the GCS url for a folder. For
+        example: `gs://my-bucket/stored. A sub-folder will be generated to
+        avoid collision between possible different concurrent runs.
+    :type gcs_staging_location: str
+    :param dbt_version: the DBT version to be fetched from dockerhub. Defaults
+        to '0.21.0'
+    :type dbt_version: str
+    :param service_account: email for the service account. If set must be
+        accompanied by the project_id
+    """
+    def __init__(
+
+        self,
+        project_id: str = None,
+        gcs_staging_location: str = None,
+        gcp_conn_id: str = "google_cloud_default",
+        dbt_version: str = '0.21.0',
+        env: Dict = None,
+        service_account=None,
+    ):
+        staging_bucket, staging_blob = _parse_gcs_url(gcs_staging_location)
+        # we have provided something similar to
+        # 'gs://<staging_bucket>/<staging_blob.tar.gz>'
```

Comment:
Because that's what cloud-build accepts (tar.gz and zip, but I opted for the first one).

You can pass over an "artifact" to cloud build that it then uncompresses